lib/foreman_maintain/package_manager/base.rb

```diff
@@ -61,6 +61,21 @@ def files_not_owned_by_package(directory)
       raise NotImplementedError
     end
 
+    # upgrade package sources to install Foreman from
+    def upgrade_foreman_repos(to_version)
```

Comment:
I did find any call to this method. 
Do you really need this? or probably can be removed.